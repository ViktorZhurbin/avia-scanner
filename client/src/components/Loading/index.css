.loader {
    display: flex;
    position: absolute;
    width: 100%;
    height: 100%;
    justify-content: center;
    align-items: center;
}

.loaderBall {
    display: inline-block;
    margin: auto 0.25rem;
    height: 0.75rem;
    width: 0.75rem;
    border-radius: 0.375rem;
    background: #06c;

    &.one,
    &.two,
    &.three {
        animation: bulging 2s infinite ease-in-out;
    }

    &.one {
        animation-delay: -0.4s;
    }

    &.two {
        animation-delay: -0.2s;
    }
}

@keyframes bulging {
    0%,
    80%,
    100% {
        transform: scale(0);
        opacity: 0.5;
    }

    40% {
        transform: scale(1);
        opacity: 1;
    }
}
