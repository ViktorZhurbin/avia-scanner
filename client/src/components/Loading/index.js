import React from 'react';
import cl from 'classnames/bind';

import styles from './index.css';

const cx = cl.bind(styles);

const Loading = () => (
    <div className={cx('loader')}>
        <span className={cx('loaderBall', 'one')} />
        <span className={cx('loaderBall', 'two')} />
        <span className={cx('loaderBall', 'three')} />
    </div>
);

export default React.memo(Loading);
