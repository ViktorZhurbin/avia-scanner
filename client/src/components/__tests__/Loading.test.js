import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '../Loading';

describe('Render Loading', () => {
    it('render correctly', () => {
        const component = renderer.create(
            <Loading />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
