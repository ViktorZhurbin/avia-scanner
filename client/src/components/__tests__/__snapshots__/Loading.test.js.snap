// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Loading render correctly 1`] = `
<div
  className="loader"
>
  <span
    className="loaderBall one"
  />
  <span
    className="loaderBall two"
  />
  <span
    className="loaderBall three"
  />
</div>
`;
